### DISCUSSION TITLE
With Android 13  Existing Camera and File Plugin Routines No Longer Function

### DISCUSSION THREAD

**Discussion by ghost** [apache/cordova#443] [Q&A]
I have an existing routine that worked fine in earlier versions that  takes a picture, renames it and moves it to a sub-directory("Photos") that  I successfully created in the sandboxed dataDirectory ("files") folder of my Application. It's no longer functioning. I cannot get the returned fileURI from the camera plugin to format correctly for the subsequent file operations. 

The photo plugin returns "fileURI " = file:///data/user/0/com.nsbasic.Vantastic/cache/1697768037972.jpg
(inside the cache folder of the apps sandbox). Simply passing the above file URI to the getfile() method returns a not found error. How can i convert the above "fileURI" to something that the getfile() can find? I have tried:

- window.resolveLocalFileSystemURL(fileURI)
- window.resolveLocalFileSystemURL(cordova.file.cacheDirectory) + "1697768037972.jpg"
- toURL(fileURI)

Any ideas?? Here is my latest attempt that fails to find the photo:

```javascript
//---Begin Photo Stuff
Mimage.onclick = function() {
    options = {
        quality: 75,
        correctOrientation: true,
        allowEdit: false,
        destinationType: 1
    };
    navigator.camera.getPicture(onPictureReturned, onPictureError, options);
};

function onPictureReturned(fileURI) {
    varLastSlash = InStrR(fileURI, "/", 0);
    varfileURI = fileURI;
    console.log(varfileURI);
    PhotoName = Mid(fileURI, varLastSlash); //The Photo name
    varPicture = Trim(gSelectedBroker) + "-" + Trim(gSelectedCategoryNumber) + "-" +             Trim(gSelectedChainNumber) + "-" + Trim(gSelectedStoreNumber) + "-" + CStr(Year(varDate)) +     PhotoMonth + PhotoDay + "-" + Trim(varQuestionNO);
varPicture = varPicture + ".jpg";
    MovePicS();
}

function onPictureError(message) {
    NSB.MsgBox("Failed because " + message);
}

function MovePicS() {
    gfileSystem.root.getFile(varfileURI, {
        create: false
    }, gotFileEntryS, fail2st);
}

function gotFileEntryS(fe) {
    FileEntry = fe;
    FileEntry.moveTo(gdataDir, varPicture, SubSuccess2, fail3st);
}

function SubSuccess2(entry) {
    //MsgBox "Successfully Moved file from Cache:" & vbCRLF & PhotoName & vbCRLF & "To Photos: " & varPicture
    NSB.MsgBox("Press [SAVE ANSWER] to Complete Process...", 0, "Photo Successfull");
}

function fail2st(error) {
    console.log(error);NSB.MsgBox("Error getFile(PhotoName: " + error.code + " -" + error.error + "-" + PhotoName);
    NSB.MsgBox("Error getFile(PhotoName: " + error.code + " -" + error.error + "-" + PhotoName);
}

function fail3st(error) {
    NSB.MsgBox("Error .moveTo(gdataDir: " + error.code + " -- " + error.error + "-" + varPicture);
}
//---End Photo Stuff
```

CHEERS!
jimmy

**Comment by breautek** [comment]
> function gotFileEntryS(fe) {
    FileEntry = fe;
    FileEntry.moveTo(gdataDir, varPicture, SubSuccess2, fail3st);
}

It's likely unrelated but `FileEntry` is an actual class created by Cordova that you might be overwriting, you shouldn't reassign the `window.FileEntry` object.

The equivalent would be

```
function gotFileEntryS(fe) {
    fe.moveTo(gdataDir, varPicture, SubSuccess2, fail3st);
}
```

I don't think your code will cause problems on the first time run, but it will likely cause problems on the second image capture, or any file "get" operation thereafter.

`window.resolveLocalFileSystemURL(fileURI)` should work on the returned file.

Can you include more details about your environment including camera and file plugin versions, and your cordova-android version?

**Reply by ghost** [reply]
Using the latest Plugins from NPM:
<plugin name="cordova-plugin-camera" source="npm" />
<plugin name="cordova-plugin-file" source="npm"  />
<preference name="android-minSdkVersion" value="30" />
<preference name="android-targetSdkVersion" value="33" />
The app worked fine for years.
I am using the internal storage. The photo is in the applications cache.  Just want to know what to do to a fileURI that is returned from your photo plugin, in order to get a file entry object getfile(fileURI...  (no longer excepts this and Evrything else I tried.. ... I GET  FILE NOT FOUND . How do I format the returned fileURI to get it in the appropriate format for getfile() operation??

**Reply by ghost** [reply]
plugin name="cordova-plugin-camera" source="npm"'
<plugin name="cordova-plugin-file" source="npm"  

preference name="android-minSdkVersion" value="30" 

preference name="android-targetSdkVersion" value="33"
